### PR TITLE
Extract credit card list item into reusable component

### PR DESCRIPTION
**Pull request by dechov**
Refactor to extract credit card list item component out of `my-sites` and into `components`.

- `my-sites/checkout/checkout/stored-card` — already being used outside of `my-sites` in `me/purchases/credit-cards` — is moved into `components/credit-card/stored-card`
- `components/credit-card` is a parent component that can either render a `<StoredCard>` if passed a `card` prop, or serve as a wrapper for passed `children` (e.g. `<NewCardForm>`). Equivalent to the now-removed `section` method of `my-sites/checkout/checkout/credit-card-selector`
- removed some unnecessary markup and duplicate markup + styling

Display changes:
- Use display name for credit card type (e.g. "American Express" instead of "amex"), generalized from existing implementation in `woocommerce/woocommerce-services/views/label-settings/label-payment-method`
- Added some margin between name and expiration to trigger wrapping before getting too close

### CODE REVIEW

**Comment by alisterscott** [comment]
Just re-adding the Needs Review label to re-run the tests

**Comment by alisterscott** [comment]
@markryall would you like to review this?

File: client/components/credit-card/stored-card.jsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+
+export const getCreditCardSummary = ( translate, type, digits ) => {
+	const supportedTypes = {
+		amex: translate( 'American Express' ),
+		discover: translate( 'Discover' ),
+		mastercard: translate( 'Mastercard' ),
```
Comment (a8ci18n) [inline client/components/credit-card/stored-card.jsx]: Hi! I've found a possible matching string that has already been translated **52** times:
`translate( 'MasterCard' )` *ES Score: 32*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/credit-card-list-item-into-reusable-component)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by dechov** [comment]
FYI this `<CreditCard>` component is used in https://github.com/Automattic/wp-calypso/pull/22504 to present a credit card selector in Store shipping label settings that's more compact and consistent with the rest of Calypso.

Open to feedback on a better way to Component-ize this.

**Comment by gwwar** [comment]
To help reduce risk I'd suggest splitting this up into three PRs:
- [ ] Add the new card component (but do not touch existing usages). (janitoral)
- [ ] Another PR to wire in the new component / delete the old one (janitorial)
- [ ] A third to add any changing enhancements to the new component (changes behavior)

This makes it much easier to review, and less risky! :) Happy to help review if you'd like to do so.

**Comment by dechov** [comment]
Thanks for the advice and offer to help review! Opened https://github.com/Automattic/wp-calypso/pull/24485.

Other branches off of that one are `update/credit-card-lists-to-use-credit-card-component` and `update/enhance-credit-card-component-summary-and-responsiveness` — I can open them now and rebase later, or wait until the dependent changes are merged.

**Pull request closed without merging**